Document/0x05h-Testing-Platform-Interaction.md

```diff
@@ -42,18 +42,24 @@ The first code block defines the new permission, which is self-explanatory. The
 </activity>
 ```
 
-Once the permission `START_MAIN_ACTIVTY` has been created, apps can request it via the `uses-permission` tag in the `AndroidManifest.xml` file. Any application granted the custom permission `START_MAIN_ACTIVITY` can then launch the `TEST_ACTIVITY`.
+Once the permission `START_MAIN_ACTIVTY` has been created, apps can request it via the `uses-permission` tag in the `AndroidManifest.xml` file. Any application granted the custom permission `START_MAIN_ACTIVITY` can then launch the `TEST_ACTIVITY`. Please note starting with api 24-28 the permission must be declared before the `<application>` tag or permission will always be denied.
 
 ```xml
+<manifest>
 <uses-permission android:name="com.example.myapp.permission.START_MAIN_ACTIVITY"/>
+        <application>
+            <activity>
+            </activity>
+        </application>
+</manifest>
 ```
 
 #### Static Analysis
 
 
 **Android Permissions**
 
-Check permissions to make sure that the app really needs them and remove unnecessary permissions. For example, the `INTERNET` permission in the AndroidManifest.xml file is necessary for an Activity to load a web page into a WebView.
+Check permissions to make sure that the app really needs them and remove unnecessary permissions. For example, the `INTERNET` permission in the AndroidManifest.xml file is necessary for an Activity to load a web page into a WebView. If a dangerous permission is needed it must be granted, you must check every time that operation is performed. 
 
```

Comment:
Can you list the permissiosn which are considered dangerous?

Reply:
List added. :-)

Reply:
✔️